controllers/grid/settings/user/form/UserForm.php

```diff
@@ -159,6 +177,24 @@ public function saveUserGroupAssignments(Request $request): void
                         'dateStart' => now(),
                     ])
                 );
+
+            // update masthead
+            // ignore reviewer role
+            $reviewerUserGroupIds = Repo::userGroup()->getArrayIdByRoleId(Role::ROLE_ID_REVIEWER, $contextId);
+            collect($this->getData('userGroupIds'))
+                ->filter(
+                    function ($userGroupId) use ($reviewerUserGroupIds) {
+                        return !in_array($userGroupId, $reviewerUserGroupIds);
+                    }
+                )
+                ->each(
+                    fn ($userGroupId) =>
+                    UserUserGroup::query()
```

Comment:
I think this is missing a check to ensure that the user-specified `userGroupIds` are in the right context.